magenta-lib/src/main/scala/magenta/input/resolver/DeploymentGraphBuilder.scala

```diff
@@ -0,0 +1,46 @@
+package magenta.input.resolver
+
+import magenta.graph.{Edge, EndNode, Graph, MidNode, Node, StartNode}
+import magenta.input.Deployment
+
+object DeploymentGraphBuilder {
+  def buildGraph(deployments: List[Deployment]): Graph[Deployment] = {
+    val edges = startEdges(deployments) ::: midEdges(deployments) ::: endEdges(deployments)
+    Graph(edges.toSet)
+  }
+
+  implicit class RichDeploymentList(deployments: List[Deployment]) {
+    def names: Set[String] = deployments.map(_.name).toSet
+    def dependencies: Set[String] = deployments.flatMap(_.dependencies).toSet
+  }
+
+  private[resolver] def startEdges(deployments: List[Deployment]): List[Edge[Deployment]] = {
+    val targetNodes = for {
+      deployment <- deployments
+      if !deployment.dependencies.exists(deployments.names.contains)
+    } yield MidNode(deployment)
+    edges(StartNode, targetNodes)
+  }
+
+  private[resolver] def midEdges(deployments: List[Deployment]): List[Edge[Deployment]] = {
```

Comment:
It relates directly to `MidNode`. This creates any edges that go from a `MidNode` to another `MidNode`.